### PR TITLE
Remove checks for/requirements on writeapi

### PR DESCRIPTION
**Pull request by andrewbogott**
writeapi is being removed

Bug: T294397

### CODE REVIEW

File: examples/basic_edit.py [hunk 1]
```diff
@@ -53,12 +52,6 @@
 history = list(image.imagehistory())
 print('History:', history)
 
-if site.writeapi:
```
Comment (lucaswerkmeister) [inline examples/basic_edit.py:56]: Probably better to keep this code and only remove the condition instead?

**Review by lucaswerkmeister — APPROVED** [review]

**Comment by AdamWill** [comment]
For the record: the upstream comment notes "Deprecated since MW 1.32", and 1.32 went out of support in 2020. So I think it should be fine to remove this now.

**Comment by AdamWill** [comment]
This LGTM. @mwclient/maintainers , anyone else want to take a look?

**Comment by RheingoldRiver** [comment]
I know of some wikis still using 1.35 but yeah 1.32 seems pretty safe to drop support for 👍

**Comment by andrewbogott** [comment]
Please take heed that I have never worked on this codebase before and have also not tested this patch! It was written in an agitated rush once I noticed that things were breaking :)

**Comment by AdamWill** [comment]
yeah, I honestly have never touched any of this code either, but it's pretty simple and matches what I would do just grepping the code for writeapi. and hey, the tests pass, so we're 70% sure it's fine! :P

**Comment by AdamWill** [comment]
OK, since nobody has any objections, let's go ahead and merge this. I'm at a conference RN, I'll try and carve out more time to get back to the 0.11 release plans after that.

**Comment by marcfrederick** [comment]
I've added the breaking change label to this, as it removes the public `Site.writeapi` attribute.

**Pull request merged**